src/ORM/Marshaller.php

```diff
@@ -211,6 +211,14 @@ public function one(array $data, array $options = [])
             $entity->set($properties);
         }
 
+        // Don't flag clean association entities as
+        // dirty so we don't persist empty records.
+        foreach ($properties as $field => $value) {
+            if ($value instanceof EntityInterface) {
+                $entity->dirty($field, $value->dirty());
```

Comment:
I'm a little worried about this change as well. I contemplated changing `merge()` so that it marked the association property as dirty if the property did not exist before but had a record marshalled. I think having merge() and one() be consistent is what I think should change.

The list of properties here are the ones that have already been set into the entity on line 211, so it should be less risky to propagate the dirty state.